src/Effects.js

```diff
@@ -274,75 +370,33 @@ module.exports = class Effects {
     };
 
     this.pineapples = {
-      pineapple: [],
-      size: 50,
+      pineappleList: [],
       init: function () {
-        for (let i = 0; i < 20; i++) {
-          this.pineapple.push({
-            x: randomNumber(20, 380),
-            y: randomNumber(20, 380),
+        for (let i = 0; i < 8; i++) {
+          this.pineappleList.push({
+            x: randomNumber(10, 390),
+            y: randomNumber(10, 390),
             rot: randomNumber(0, 359),
-            speed: randomNumber(2, 5),
+            life: 5,
           });
         }
       },
-      update: function () {
-        this.size += randomNumber(-5, 5);
-      },
       draw: function () {
-        if (this.pineapple.length < 1) {
+        if (this.pineappleList.length < 1) {
           this.init();
         }
-        for (let i = 0; i < this.pineapple.length; i++) {
+        for (let i = 0; i < this.pineappleList.length; i++) {
           p5.push();
-          const pineapple = this.pineapple[i];
+          const pineapple = this.pineappleList[i];
           p5.translate(pineapple.x, pineapple.y);
           p5.rotate(pineapple.rot);
-          p5.scale(2, 2);
+          p5.scale(pineapple.life / 20);
           drawPineapple(p5._renderer.drawingContext);
```

Comment:
Pineapples was identified as one of the slow backgrounds in https://github.com/code-dot-org/dance-party/issues/452.  It looks like you chose not to use the draw-once-then-copy strategy here.  Is there a particular reason why?

Reply:
great catch; i will clean these up.